06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * An inverted-index class
+ * @class
+ */
+class InvertedIndex {
+  /**
+  * class constructor
+  * @constructor
+  */
+  constructor() {
+    this.allIndex = {};
+    this.allFiles = {};
+    this.allLength = {};
+    this.allFilesTitle = [];
+    this.validatedFile = {};
+  }
+
+  /**
+   * Validate Method
+   * @param {Object} validateFile an object to be validated
+   * @returns {Object} file that has been validated
+   */
+  validateFile(validateFile) {
```

Comment:
You still cannot name your variable `validateFile`.

Also, think of how you can simplify this entire function using a ternary operator.